arangod/IResearch/IResearchView.cpp

```diff
@@ -1935,30 +1954,40 @@ bool IResearchView::sync(size_t maxMsec /*= 0*/) {
   return type;
 }
 
-arangodb::Result IResearchView::updateLogicalProperties(
-  arangodb::velocypack::Slice const& slice,
-  bool partialUpdate,
-  bool doSync
-) {
-  return _logicalView
-    ? _logicalView->updateProperties(slice, partialUpdate, doSync)
-    : arangodb::Result(TRI_ERROR_ARANGO_DATA_SOURCE_NOT_FOUND)
-    ;
+void IResearchView::toVelocyPack(
+    velocypack::Builder& result,
+    bool includeProperties,
+    bool includeSystem
+) const {
+  // We write into an open object
+  TRI_ASSERT(result.isOpenObject());
+
+  DBServerLogicalView::toVelocyPack(result, includeProperties, includeSystem);
+
+  // Object is still open
+  TRI_ASSERT(result.isOpenObject());
+
+  if (includeProperties) {
+    // implementation Information
+    result.add("properties", VPackValue(VPackValueType::Object));
+    // note: includeSystem and forPersistence are not 100% synonymous,
+    // however, for our purposes this is an okay mapping; we only set
+    // includeSystem if we are persisting the properties
+    getPropertiesVPack(result, includeSystem);
+    result.close();
+  }
+
+  TRI_ASSERT(result.isOpenObject()); // We leave the object open
 }
 
 arangodb::Result IResearchView::updateProperties(
-  arangodb::velocypack::Slice const& slice,
-  bool partialUpdate,
-  bool doSync
+    const velocypack::Slice &slice,
```

Comment:
At some stage we had the coding guideline to use `velocypack::Slice const&` here as type name. It is a more consistent way to write things because then `const` always applies to the thing directly in front of it. We usually group the `*` with the type and not with the identifier.

Reply:
done